From 61d4b099fac1d54704a3021ee3919111f2e91bd7 Mon Sep 17 00:00:00 2001
From: "J. Michael Welsh" <mike.welsh@arm.com>
Date: Tue, 22 Sep 2020 19:15:48 +0000
Subject: [PATCH] Fix CPU Temp Path

Changing the CPU temperature path to work with the hardware for this
build.
---
 edge-client/gateway_services_resource.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/edge-client/gateway_services_resource.c b/edge-client/gateway_services_resource.c
index cf424b3..eeb9edb 100644
--- a/edge-client/gateway_services_resource.c
+++ b/edge-client/gateway_services_resource.c
@@ -352,7 +352,7 @@ float get_cpu()
 void *gsr_update_gateway_stats_resource(void *args)
 {
     // CPU temperature in Celsius
-    const char cmd_cpu_temp[] = "{ echo -1; cat /sys/class/hwmon/hwmon0/temp*_input 2>/dev/null; } | awk '{if (max<$1) max=$1} END {print max/1000}'";
+    const char cmd_cpu_temp[] = "{ echo -1; cat /sys/class/thermal/thermal_zone0/temp 2>/dev/null; } | awk '{if (max<$1) max=$1} END {print max/1000}'";
 
     // RAM free and total in bytes
     const char cmd_ram_free[] = "awk '/^MemFree:/{ print $2*1024 }' /proc/meminfo";
-- 
2.20.1

